examples/cbindgen/cbindgen_example.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    ffi::{CStr, CString},
+    os::raw::c_char,
+    f32::consts::PI,
+};
+
+extern crate libc;
+use libc::size_t;
+
+#[no_mangle]
+pub extern "C" fn int_from_rust() -> i32 {
+    1337
+}
+
+#[no_mangle]
+pub extern "C" fn int_into_rust(num: i32) {
+    assert_eq!(num, 9999);
+}
+
+/** # Safety
+ * 
+ * Dereferencing the pointer `buffer` is not safe
+ */
+#[no_mangle]
+pub unsafe extern "C" fn string_from_rust(buffer: *mut *mut c_char) -> size_t {
+    let text = "Hello Fellow Rustacean!";
+    let c_text: CString = CString::new(text).unwrap();
+    *buffer = c_text.into_raw();
+
+    text.len() as size_t
+}
+
+/** # Safety
+ * 
+ * Dereferencing the pointer `buffer` is not safe
+ */
+#[no_mangle]
+pub unsafe extern "C" fn free_rust_string(buffer: *mut c_char) {
+    // The constructed CString will free the memory
+    CString::from_raw(buffer);
+}
+
+/** # Safety
+ * 
+ * Dereferencing the pointer `text` is not safe
+ */
+#[no_mangle]
+pub unsafe extern "C" fn string_into_rust(text: *const c_char) {
+    let string = CStr::from_ptr(text);
+    assert_eq!(string.to_str().unwrap(), "Hello World!");
+}
+
+#[repr(C)]
+pub struct Point {
+    pub x: f32, 
+    pub y: f32,
+}
+
+#[no_mangle]
+pub extern "C" fn struct_from_rust() -> Point {
+    Point {
+        x: PI,
+        y: 42.0f32,
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn struct_into_rust(point: Point) {
+    let epsilon = 0.01f32;
+    assert!((point.x - 42.0f32).abs() < epsilon);
```

Comment:
Out of curiosity, shouldn't the result be bitwise identical?

Reply:
It is but clippy was complaining about float comparisons so I (likely did the bad thing and) changed the implementation. Any suggestions?

Reply:
Looks like you can skip lints, https://stackoverflow.com/questions/55402812/how-to-disable-a-clippy-lint-for-a-single-line-block

Reply:
If this is for https://rust-lang.github.io/rust-clippy/master/#float_cmp , there are some examples of how to fix.

@UebelAndre , are you still running into issues with clippy beyond the examples directory?

Reply:
No, that was the only outstanding issue. I had some complaints about the use of `unsafe` and the `float_cmp`. I'll check those out, thanks guys 😄